test/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/RemoveRecordControllerSpec.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.controllers
+
+import cats.data.EitherT
+import org.mockito.ArgumentMatchersSugar.any
+import org.mockito.MockitoSugar.when
+import org.scalatestplus.mockito.MockitoSugar
+import org.scalatestplus.play.PlaySpec
+import play.api.http.Status.{BAD_REQUEST, INTERNAL_SERVER_ERROR, OK}
+import play.api.libs.json.Json
+import play.api.mvc.Results.InternalServerError
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{contentAsJson, defaultAwaitTimeout, status, stubControllerComponents}
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.errors.{Error, ErrorResponse}
+import uk.gov.hmrc.tradergoodsprofilesrouter.service.{RouterService, UuidService}
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.{ApplicationConstants, HeaderNames}
+
+import scala.concurrent.ExecutionContext
+
+class RemoveRecordControllerSpec extends PlaySpec with MockitoSugar {
+
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  val mockRouterService = mock[RouterService]
+  val mockUuidService   = mock[UuidService]
+
+  private val sut =
+    new RemoveRecordController(
+      stubControllerComponents(),
+      mockRouterService,
+      mockUuidService
+    )
+
+  def validHeaders: Seq[(String, String)] = Seq(
+    HeaderNames.ClientId -> "clientId"
+  )
+
+  "PUT /:eori/records/:recordId" should {
+
+    "return a 200 Ok response on removing a record" in {
+
+      when(mockRouterService.removeRecord(any, any, any)(any, any))
+        .thenReturn(EitherT.rightT(OK))
+
+      val result = sut.remove("GB123456789001", "12345")(
+        FakeRequest().withBody(removeRecordRequestData).withHeaders(validHeaders: _*)
+      )
+
+      status(result) mustBe OK
+    }
+    "return 400 Bad request when mandatory request header X-Client-ID" in {
+
+      when(mockUuidService.uuid).thenReturn("8ebb6b04-6ab0-4fe2-ad62-e6389a8a204f")
+      val result = sut.remove("GB123456789001", "12345")(
+        FakeRequest().withBody(removeRecordRequestData)
+      )
+      status(result) mustBe BAD_REQUEST
+      contentAsJson(result) mustBe Json.toJson(createErrorResponse)
+    }
+    "return 400 Bad request when required request field actorId is missing" in {
+      val errorResponse = ErrorResponse(
```

Comment:
this could be name expectedResponse for more clarity. Also in test I think it is better to use literal instead on production code constant if you want your test be robust. For example in this case the code is testing for code to be "BAD_REQUEST" and  message to be "Bad Request". If put the case we change the constant "ApplicationConstants.BadRequestCode" and "ApplicationConstants.BadRequestMessage" to something wrong (a bad merge or spelling mistake) the test will never catch that and it will pass for the wrong reason.